server/tests/security/test_oauth_callback_redaction.py

```
@@ -0,0 +1,251 @@
+    """
+
+    @pytest.fixture(autouse=True)
+    def _setup(self, monkeypatch):
+        for pkg in _ATLASSIAN_HEAVY:
+            sys.modules.setdefault(pkg, MagicMock())
+        _install_audit_stub()
+        limiter_stub = sys.modules["utils.web.limiter_ext"]
+        limiter_stub.limiter = MagicMock()
+        limiter_stub.limiter.limit = lambda *a, **kw: (lambda f: f)
+
+    def _build_app(self, monkeypatch):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_build_app`

(ANN202)